index.d.ts

```diff
@@ -0,0 +1,25 @@
+declare function whichpm (pkgPath: string): Promise<whichpm.Result>
+
+declare namespace whichpm {
+  type Result = NPM | YARN | PNPM | ScopedPNPM
+
+  interface NPM {
+    readonly name: 'npm'
+  }
+
+  interface YARN {
+    readonly name: 'yarn'
+  }
+
+  interface PNPM {
+    readonly name: 'pnpm'
+    readonly version: string
+  }
+
+  interface ScopedPNPM {
+    readonly name: '@zkochan/pnpm'
+    readonly version: string
```

Comment:
I think it should be something more generic. like {name: string} | {name:string, version: string}

Reply:
How about this?

```typescript
interface ScopedPNPM {
  readonly name: string
  readonly version?: string
}
```

Reply:
ok, but maybe call it just OtherPM

Reply:
It will be done.